pkg/envoy/xds/ack.go

```diff
@@ -210,7 +253,53 @@ func (m *AckingResourceMutatorWrapper) HandleResourceVersionAck(version uint64,
 
 		// Completion didn't match or is still waiting for some ACKs. Keep it
 		// in the pending list.
-		remainingCompletions = append(remainingCompletions, comp)
+		remainingCompletions[comp.completion] = comp
+	}
+
+	m.pendingCompletions = remainingCompletions
+}
+
+func (m *AckingResourceMutatorWrapper) HandleResourceVersionNack(version uint64, node *envoy_api_v2_core.Node, resourceNames []string, typeURL string) {
+	ackLog := log.WithFields(logrus.Fields{
+		logfields.XDSVersionInfo: version,
+		logfields.XDSClientNode:  node,
+		logfields.XDSTypeURL:     typeURL,
+	})
+
+	nodeID, err := m.nodeToID(node)
+	if err != nil {
+		// Ignore NACKs from unknown or misconfigured nodes which have invalid
+		// node identifiers.
+		ackLog.Warning("invalid ID in Node identifier; ignoring NACK")
```

Comment:
invalid -> Invalid

Reply:
Fixed.

Reply:
There were two of these, will fix this one too.